R/case.R

```diff
@@ -54,3 +74,50 @@ str_to_sentence <- function(string, locale = "en") {
     opts_brkiter = stri_opts_brkiter(type = "sentence", locale = locale)
   )
 }
+#' @export
+#' @rdname case
+str_to_pascal <- function(string, locale = "en") {
+  stopifnot(is.character(string))
+  string <- string |>
+    str_replace_all("([a-z])([A-Z])", "\\1 \\2") |>
+    str_replace_all("([0-9])([a-zA-Z])", "\\1 \\2") |>
+    str_replace_all("([A-Z]+)([A-Z][a-z])", "\\1 \\2") |>
+    str_replace_all(pattern = "[:punct:]", replace = " ") |>
+    str_to_title(locale = locale) |>
+    str_remove_all(pattern = "\\s+")
+  return(string)
+}
+#' @export
+#' @rdname case
+StrToPascal <- str_to_pascal
+#' @export
+#' @rdname case
+str_to_camel <- function(string, locale = "en") {
+  string <- str_to_pascal(string, locale = locale)
+  string <- str_replace(string, pattern = "^.", replace = str_to_lower(str_sub(string, 1, 1)))
+  return(string)
+}
+#' @export
+#' @rdname case
+strToCamel <- str_to_camel
+#' @export
+#' @rdname case
+str_to_snake <- function(string, separator = "_", locale = "en") {
+  stopifnot(is.character(string))
+  string <- string |>
+    str_replace_all("([a-z])([A-Z])", "\\1 \\2") |>
+    str_replace_all("([a-zA-Z])([0-9])", "\\1 \\2") |>
```

Comment:
I think it would make sense to extract out this code for "break into words" into a separate function since it's reasonably complex and giving it a name would be useful.